asciimatics/renderers/charts.py

```diff
@@ -0,0 +1,486 @@
+    #: Constant to indicate no axes should be rendered.
+    NONE = 0
+    NO_AXIS = 0
+
+    #: Constant to indicate just the x axis should be rendered.
+    X_AXIS = 1
+
+    #: Constant to indicate just the y axis should be rendered.
+    Y_AXIS = 2
+
+    #: Constant to indicate both axes should be rendered.
+    BOTH = 3
+    BOTH_AXES = 3
+
+    def __init__(self, height, width, functions, char="#", colour=Screen.COLOUR_GREEN,
+                 bg=Screen.COLOUR_BLACK, gradient=None, scale=None, axes=Y_AXIS, intervals=None,
+                 labels=False, border=True, keys=None, gap=None):
+        ### See children BarChart and VBarChart for argument descriptions and pydocs
+        super(_BarChartBase, self).__init__(height, width)
+        self._functions = functions
+        self._char = char
+        self._colours = [colour] if isinstance(colour, int) else colour
+        self._bgs = [bg] if isinstance(bg, int) else bg
+        self._scale = scale
+        self._axes = axes
+        self._intervals = intervals
+        self._labels = labels
+        self._border = border
+        self._keys = keys
+        self._gap = gap
+
+        # Box drawing tool for border, allows user to change the border line style
+        self._border_lines = BoxTool(self._canvas.unicode_aware, DOUBLE_LINE) if border else None
+
+        # Box drawing tool for axes
+        self._axes_lines = BoxTool(self._canvas.unicode_aware, SINGLE_LINE)
+
+        # Normalize the gradient so that it is 3-tuple wide (bg is optional, if not there, set it)
+        self._gradient = None
+        if gradient:
+            self._gradient = []
+            for item in gradient:
+                if len(item) == 2:
+                    self._gradient.append( (item[0], item[1], Screen.COLOUR_BLACK) )
+                elif len(item) == 3:
+                    self._gradient.append(item)
+                else:
+                    raise ValueError("Gradients must be 2-tuple or 3-tuple in size")
+
+    @abstractmethod
+    def _render_now(self):
+        pass
+
+    @property
+    def line_style(self):
+        """Returns the current drawing style of the border and axes. Possible values are defined in
+        :mod:`~asciimatics.constants`:
+
+        * `ASCII_LINE` -- ASCII safe characters
+        * `SINGLE_LINE` -- UNICODE based single line
+        * `DOUBLE_LINE` -- UNICODE based double line
+
+        Note that your canvas must support UNICODE style characters to use them
+
+        Single and double-line styles use a single-line axes. The ASCII style uses ASCII for the
+        border and the axes.
+        """
+        return self._border_lines.style
+
+    @line_style.setter
+    def line_style(self, style):
+        if self._border_lines:
+            self._border_lines.style = style
+
+            if style == DOUBLE_LINE:
+                self._axes_lines.style = SINGLE_LINE
+            else:
+                self._axes_lines.style = style
```

Comment:
I must have misunderstood a previous conversation on this, thought this was what you wanted.

So, no axis style at all? Always use ASCII characters?

I like the unicode stuff, but don't have a strong opinion here. Your library.

Reply:
Yeah - this gets a bit subtle.  I missed the fact that you have added a feature to the BarChart, by making it capable of handling unicode stuff now.  It's been too lon since I wrote the original!  :-)

Ideally, we'd just make the renderer unicode aware, which would mean adding defaulted parameters to TemporaryCanvas and the Chart classes.  This can then be passed in to either BoxTool object when setting a style.  We can then set a default style for both the frame and the axes - which will work as expected on unicode/ascii screens.

At that point, I don't feel we really need to expose the axis setting at all.

Make sense?

Reply:
Let me make sure I understand the ask:

* border is exposed through the style property
* changes to border style property only effect border
* border and axes lines default based on canvas unicode detect, only using ASCII if the canvas can't do unicode
* if the user for some reason changes the border style to ASCII even if unicode is possible, the axes stays unicode

Correct?

Reply:
Yeah - I honestly don't expect people to want to use ASCII when they don't need to.  If someone really wants to do that, we just expose the separate property for the axes lines.

Reply:
Though, if you want to add the axis property too, I'm happy to take it.  Just trying to minimize your work...

Reply:
Oh I re-read and I'm not advocating for extra checks here...  if a user asks for ascii borders on a unicode capable canvas, we should not override.  Just let BoxTool use its existing logic.

Reply:
Ok. Went back to using the property. Not a lot of code, have to have the 2 BoxTools anyway (can't have the axes be double when the border is).